File: mpp/lab3/src/main/java/repository/IScoreRepository.java
package repository;

import domain.Score;
import domain.ScoreType;

import java.util.List;

public interface IScoreRepository {
    void addScore(int participantId, ScoreType type, int score) throws RepositoryError;
    void updateScore(int participantId, ScoreType type, int score) throws RepositoryError;
    void setScore(int participantId, ScoreType type, int score) throws RepositoryError;
    List<Score> findScoresForTypeSortedDescending(ScoreType type) throws RepositoryError;
    List<Score> findScoresSortedByName() throws RepositoryError;
}

File: mpp/lab7/server/Repository/IScoreRepository.cs
using System.Collections.Generic;
using server.Domain;

namespace server.Repository
{
    public interface IScoreRepository
    {
        void Add(int participantId, ScoreType type, int value);
        Score Get(int participantId);
        void Update(int participantId, ScoreType type, int value);
        void Set(int participantId, ScoreType type, int value);
        List<Score> FindByTypeSortedDescending(ScoreType type);
        List<Score> FindSortedByName();
    }
}
